File: E-Tour/src/main/java/com/example/demo/entities/SubCategoryMaster.java
package com.example.demo.entities;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "sub_category_master")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class SubCategoryMaster {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "subcat_id")
    @EqualsAndHashCode.Include
    private Integer subcatId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cat_id")
    @ToString.Exclude
    private Category category;

    @Column(name = "subcat_name", length = 100)
    private String subcatName;

    @Column(name = "subcat_image_path", length = 255)
    private String subcatImagePath;

    @Column(name = "flag")
    private Boolean flag;

    @Column(name = "isactive")
    private Boolean isactive;

    @OneToMany(mappedBy = "subCategory", fetch = FetchType.LAZY)
    @Builder.Default
    @ToString.Exclude
    private List<Tour> tours = new ArrayList<>();
}

File: ETour-DotNet/Models/SubCategoryMaster.cs
namespace ETour.Api.Models;

public class SubCategoryMaster
{
    public int SubcatId { get; set; }

    public int? CatId { get; set; }
    public Category Category { get; set; }

    public string SubcatName { get; set; }
    public string SubcatImagePath { get; set; }
    public bool? Flag { get; set; }
    public bool? Isactive { get; set; }

    public ICollection<Tour> Tours { get; set; } = new List<Tour>();
}
